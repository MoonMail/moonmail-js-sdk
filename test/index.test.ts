import MoonMail from '../src';

describe('sdk', () => {
  let mm: MoonMail;
  beforeEach(() => {
    mm = new MoonMail({accountId: '12345', contact: {address: 'email@example.com'}});
    mm.request = jest.fn();
  });

  it('should updateContact', () => {
    mm.updateContact({attributes: {foo: 'bar'}, metrics: {count: 1}});
    expect(mm.request).toMatchSnapshot();
  });

  it('should triggerEvent', () => {
    mm.triggerEvent('TestEvent');
    mm.triggerEvent({eventType: 'TestEvent'});
    mm.triggerEvent(['TestEvent1', 'TestEvent2'])
    mm.triggerEvent([{eventType: 'TestEvent1'}, {eventType: 'TestEvent2'}])
    mm.triggerEvent({eventType: 'TestEvent', attributes: {foo: 'bar'}, metrics: {count: 1}})
    expect(mm.request).toMatchSnapshot();
  });
});
