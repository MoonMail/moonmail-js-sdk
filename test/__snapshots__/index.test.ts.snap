// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sdk should triggerEvent 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "post",
      "events",
      Object {
        "Contact": Object {
          "Address": "email@example.com",
        },
        "Events": Array [
          Object {
            "EventType": "TestEvent",
          },
        ],
      },
    ],
    Array [
      "post",
      "events",
      Object {
        "Contact": Object {
          "Address": "email@example.com",
        },
        "Events": Array [
          Object {
            "EventType": "TestEvent",
          },
        ],
      },
    ],
    Array [
      "post",
      "events",
      Object {
        "Contact": Object {
          "Address": "email@example.com",
        },
        "Events": Array [
          Object {
            "EventType": "TestEvent1",
          },
          Object {
            "EventType": "TestEvent2",
          },
        ],
      },
    ],
    Array [
      "post",
      "events",
      Object {
        "Contact": Object {
          "Address": "email@example.com",
        },
        "Events": Array [
          Object {
            "EventType": "TestEvent1",
          },
          Object {
            "EventType": "TestEvent2",
          },
        ],
      },
    ],
    Array [
      "post",
      "events",
      Object {
        "Contact": Object {
          "Address": "email@example.com",
        },
        "Events": Array [
          Object {
            "Attributes": Object {
              "foo": "bar",
            },
            "EventType": "TestEvent",
            "Metrics": Object {
              "count": 1,
            },
          },
        ],
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`sdk should updateContact 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "post",
      "contacts",
      Object {
        "Address": "email@example.com",
        "Attributes": Object {
          "foo": "bar",
        },
        "Metrics": Object {
          "count": 1,
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
